src/components/user/sign-up/sign-up-form.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { reduxForm } from 'redux-form';
+import asyncValidate from '../../common/form/asyncValidate';
+import Button from '../../common/form/button';
+import TextInput from '../../common/form/input-fields/text-input';
+
+class SignUpForm extends React.PureComponent {
```

Comment:
Abstract form to an HOC component so it can be reused to create other forms.

Reply:
Hey Howard. I am gonna tackle this on the next PR. I have opened a ticket for this https://github.com/AntonioLok/online-shopping-reactJS/projects/1. I have so many changes right now and I dont want to overwhelm you with so many things so its better if I tackle this on the next PR.